src/CommonLib/LDAPUtils.cs

```diff
@@ -483,22 +507,24 @@ public string GetSidFromDomainName(string domainName)
         public IEnumerable<string> DoRangedRetrieval(string distinguishedName, string attributeName)
         {
             var domainName = Helpers.DistinguishedNameToDomain(distinguishedName);
-            var task = Task.Run(() => CreateLDAPConnection(domainName, authType: _ldapConfig.AuthType));
+            var task = Task.Run(() => CreateLDAPConnectionWrapper(domainName, authType: _ldapConfig.AuthType));
 
-            LdapConnection conn;
+            LdapConnectionWrapper connWrapper;
 
             try
             {
-                conn = task.ConfigureAwait(false).GetAwaiter().GetResult();
+                connWrapper = task.ConfigureAwait(false).GetAwaiter().GetResult();
             }
             catch
             {
                 yield break;
             }
 
-            if (conn == null)
+            if (connWrapper == null)
```

Comment:
Is this an error condition if the connection wrapper is `null`?

Reply:
Good catch, this should actually be checking the connection inside the wrapper, as the wrapper can never be null